### ISSUE TITLE
[en] hadn't drank -> hadn't drunk

### ISSUE THREAD

**Issue by Tex2002ans** [languagetool-org/languagetool#4365]
This sentence is correct:

> It felt like I **hadn't drank** anything for a week.

"hadn't drunk" is wrongly recommended.

(Correct) And this equivalent sentence finds no errors:

> It felt like I had not drank anything for a week.

**Comment by MikeUnwalla** [comment]
In British English, "hadn't drank" is not correct. We say "hadn't drunk".

Also, one suggestion is the same as the incorrect text:
![image](https://user-images.githubusercontent.com/5816160/108175358-94779f80-70f8-11eb-9371-445cd5e36556.png)

**Comment by udomai** [comment]
> This sentence is correct:
> 
> > It felt like I **hadn't drank** anything for a week.
> 
> "hadn't drunk" is wrongly recommended.

I agree with @MikeUnwalla, "drank" may be acceptable regionally and/or colloquially (there is a colloquial tendency of replacing past participles with the simple past stem). But in formal, standard language, "drank" is not a past participle ([Merriam Webster](https://www.merriam-webster.com/dictionary/drink), the most descriptive one, acknowledges it, but [Cambridge](https://dictionary.cambridge.org/dictionary/english/drink) and [Oxford](https://www.lexico.com/definition/drink) do not).

@htft, what is your take on this one?

**Comment by Tex2002ans** [comment]
> In British English, "hadn't drank" is not correct. We say "hadn't drunk".
> 
> Also, one suggestion is the same as the incorrect text:
> ![image](https://user-images.githubusercontent.com/5816160/108175358-94779f80-70f8-11eb-9371-445cd5e36556.png)

I had the troublesome sentence sitting around for years.

In an older version of LT, this sentence was an infinite loop—LT would recommend "hadn't drunk" <-> "hadn't drank" forever.

Anyway, in the US, I've only heard "had not drank" or "hadn't drank". It may be a regional thing as well... unsure. This is a tough one!

**Comment by Tex2002ans** [comment]
Looks like someone did a lot of research on this "hadn't drank/drunk" situation:

https://english.stackexchange.com/questions/70343/usage-question-i-hadnt-drank-any-coffee-before-i-lived-in-italy

> I have run into a native speaker using the past form of the verb (*drank*) where the grammar calls for the past participle (*drunk*).
>
> For example: "I hadn't **drank** any coffee before I lived in Italy."
>
> According to grammar norms, this sentence should be: "I hadn't **drunk** any coffee..."
> 
> More and more, I noticed that people tend to use *drank* instead of *drunk* after perfect aspect constructions. This construction was used across all communities of practice and wasn't subject to dialect boundaries.
>
> According to results from [http://www.phras.in](http://www.phras.in), have drank and have drunk are clear, **with the standard have drunk being 4x more prevalent than have drank**. However, with an instance rate of about 200,000 for the non-standard form, that is nothing to scoff at. However, **the opposite was true with the negative form haven't drank vs. haven't drunk with the non-standard being almost twice as prevalent**.

Skimming through the comments, it seems like this could also be a Northeast US dialect thing... hmmmmm...

- - -

He also links to a paper:

- ["I haven't drank in weeks: The use of past tense forms as past participles in English corpora" by Kristina Geeraert and John Newman](https://www.researchgate.net/publication/263299763_I_haven't_drank_in_weeks_The_use_of_past_tense_forms_as_past_participles_in_English_corpora) (Language and Computers 73)

(I haven't read this yet.)
